nflow-rest-api-spring-web/src/main/java/io/nflow/rest/v1/springweb/WorkflowInstanceResource.java

```diff
@@ -93,60 +103,94 @@ public WorkflowInstanceResource(SchedulerService scheduler, WorkflowInstanceServ
   }
 
   @GetMapping
-  @ApiOperation(value = "List workflow instances", response = ListWorkflowInstanceResponse.class, responseContainer = "List")
+  @Operation(summary = "List workflow instances")
+  @ApiResponse(responseCode = "200",
+      content = @Content(array = @ArraySchema(schema = @Schema(implementation = ListWorkflowInstanceResponse.class))))
   public Mono<ResponseEntity<?>> listWorkflowInstances(
-      @RequestParam(value = "id", defaultValue = "") @ApiParam("Internal id of workflow instance") List<Long> ids,
-      @RequestParam(value = "type", defaultValue = "") @ApiParam("Workflow definition type of workflow instance") List<String> types,
-      @RequestParam(value = "parentWorkflowId", required = false) @ApiParam("Id of parent workflow instance") Long parentWorkflowId,
-      @RequestParam(value = "parentActionId", required = false) @ApiParam("Id of parent workflow instance action") Long parentActionId,
-      @RequestParam(value = "state", defaultValue = "") @ApiParam("Current state of workflow instance") List<String> states,
-      @RequestParam(value = "status", defaultValue = "") @ApiParam("Current status of workflow instance") List<WorkflowInstanceStatus> statuses,
-      @RequestParam(value = "businessKey", required = false) @ApiParam("Business key for workflow instance") String businessKey,
-      @RequestParam(value = "externalId", required = false) @ApiParam("External id for workflow instance") String externalId,
-      @RequestParam(value = "stateVariableKey", required = false) @ApiParam("Key of state variable that must exist for workflow instance") String stateVariableKey,
-      @RequestParam(value = "stateVariableValue", required = false) @ApiParam("Current value of state variable defined by stateVariableKey") String stateVariableValue,
-      @RequestParam(value = "include", required = false) @ApiParam(value = INCLUDE_PARAM_DESC, allowableValues = INCLUDE_PARAM_VALUES, allowMultiple = true) String include,
-      @RequestParam(value = "queryArchive", required = false, defaultValue = QUERY_ARCHIVED_DEFAULT_STR) @ApiParam("Query also the archive if not enough results found from main tables") boolean queryArchive,
-      @RequestParam(value = "maxResults", required = false) @ApiParam("Maximum number of workflow instances to be returned") Long maxResults,
-      @RequestParam(value = "maxActions", required = false) @ApiParam("Maximum number of actions returned for each workflow instance") Long maxActions) {
-    return handleExceptions(() -> wrapBlocking(
-        () -> ok(super.listWorkflowInstances(ids, types, parentWorkflowId, parentActionId, states, statuses, businessKey,
-            externalId, stateVariableKey, stateVariableValue, include, maxResults, maxActions, queryArchive, this.workflowInstances,
-            this.listWorkflowConverter).iterator())));
+      @RequestParam(value = "id", defaultValue = "") @Parameter(description = "Internal id of workflow instance") Set<Long> ids,
```

Comment:
Most of the other RequestParams have required=false, but id does not have? does the defaultValue imply required=false?

Reply:
fixed